ice.c

```diff
@@ -3183,6 +3227,78 @@ void *janus_ice_send_thread(void *data) {
 					}
 				}
 			}
+			if (stream && stream->do_transport_wide_cc) {
+				/* Create a transport wide feedback message*/
+				size_t size = 1300;
+				char rtcpbuf[1300];
+				
+				/* Lock session */
+				janus_mutex_lock(&handle->stream->mutex);
+				
+				/* Order packet list */
+				GSList* sorted = g_slist_sort(handle->stream->transport_wide_received_seq_nums,rtcp_transport_wide_cc_stats_comparator);
+				
+				/* Create full stats queue */
+				GQueue* packets = g_queue_new();
+				
+				/* For all packets */
+				for (GSList *it = sorted; it; it = it->next)
```

Comment:
Just thinking out loud here, so please correct me if I'm wrong, but rather than iterating on a possibly long list for every feedback message (something we've removed for other features we had, as from code profiling it's where we took most of the time) a possible alternative might be working on the RTCP message packet by packet: i.e., the RTCP message is part of the stream struct, and every time we get a new RTP packet, we update the RTCP message itself as you do here rather than iterating, and when it's time to send it, we just send what we prepared over time. This would save us the need to allocate and keep a list, and iterating over it at specified instants: we can just reset the RTCP message when we're done, and always work on the same one. Does this make sense?

Reply:
UPDATE: seeing how you pass the queue to create the message, we might use that as the part we update packet by packet, instead of storing a list to update the queue: I guess this would still do the trick and help make it more efficient by removing the traversal (or at least A traversal).

Reply:
The problem are out of order packets, as you need to provide the difference with previous one. So if you  receive packet n, misses n+1, and receives n+2, then you will calculate the delta for n+2 since n. But if you receive later n+1, the delta for n+2 would be incorrect.

Reply:
Ack, thx for clarifying!